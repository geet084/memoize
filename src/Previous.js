import React, { Component } from 'react';
import './CSS/Main.scss';

export default class Previous extends Component {

  render() {
    let {prevAnswer } = this.props;
    let answer;
    prevAnswer ? answer = 'correct' : answer = 'incorrect'

    return (
      <div>
        <p className="previous">
          Previous result for this question was:
          <span className={prevAnswer.toString()}> {answer}</span>
        </p>
      </div>
    )
  }
}