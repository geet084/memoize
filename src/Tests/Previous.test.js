import React from 'react';
import Previous from '../Previous.js';
import { shallow } from 'enzyme';

const mockPrevAnswer = true;

describe('Previous', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Previous prevAnswer={mockPrevAnswer} />
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ prevAnswer: false })
    
    expect(wrapper).toMatchSnapshot();
  })

});

