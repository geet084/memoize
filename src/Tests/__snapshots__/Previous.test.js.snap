// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Previous should match the snapshot with all data passed in correctly 1`] = `
<div>
  <p
    className="previous"
  >
    Previous result for this question was:
    <span
      className="true"
    >
       
      correct
    </span>
  </p>
</div>
`;

exports[`Previous should match the snapshot with all data passed in correctly 2`] = `
<div>
  <p
    className="previous"
  >
    Previous result for this question was:
    <span
      className="false"
    >
       
      incorrect
    </span>
  </p>
</div>
`;
